### PR TITLE
Feature/personal page

### LINKED ISSUE #364: Personal page
## Motivations
Create a personal page as in the [design](https://france-ioi.github.io/algorea-designs/01.Groups_06/Personal-data) or [design branch (yourself in the left meny)](http://dev.algorea.org/branch/design)

## Subtasks
* The page will be accessible through the top right menu > "profile" (on `/groups/me`)
* For now, only display:
  * the title (firstname, lastname).
  * the content of the "personal data" tab. (without the tabs) (could already be in a dedicated component)
  * the info in the design and that you can find in the service `https://france-ioi.github.io/algorea-devdoc/api/#operation/userData` 
  * no "modify" buttons for now.
* No need to fetch the data, it has already been loaded by `UserSessionService`.
* While adding new attributes to CurrentUserHttpService, you can convert this service to decoders as done in `GetItemByIdService`.

### PR DESCRIPTION
**Pull request by Iloveall**
Fixes #364

### CODE REVIEW

File: src/app/shared/http-services/current-user.service.ts [hunk 1]
```diff
@@ -34,12 +50,11 @@ export class CurrentUserHttpService {
       .get<RawUserProfile>(`${appConfig().apiUrl}/current-user`)
       .pipe(
         map(raw => ({
+          ...raw,
```
Comment (smadbe) [inline src/app/shared/http-services/current-user.service.ts:53]: Check get-item-by-id and get-group-by-id. The get should be to an `unknown` type and then just decode.
Stick to the attribute names used in the backend, so use `groupId` and `tempUser` everywhere (may require to change other places of the code).

File: src/app/modules/group/pages/current-user/current-user.component.html [hunk 2]
```diff
@@ -1 +1,81 @@
-<div>Yourself page works</div>
+<ng-container *ngIf="currentUser$ | async as currentUser">
+  <div class="alg-page-header">
+    <div class="alg-page-header-title">{{ currentUser.firstName }} {{ currentUser.lastName }}</div>
+  </div>
```
Comment (smadbe) [inline src/app/modules/group/pages/current-user/current-user.component.html:4]: Could you try to have the page title a bit larger than section title? A bit like [here](https://france-ioi.github.io/algorea-designs/01.Groups_06/Personal-data). Of course, we'll have to check that it is ok on the other pages as well, but that should make it.

File: src/app/modules/group/pages/current-user/current-user.component.ts [hunk 3]
```diff
@@ -9,14 +11,20 @@ const currentUserBreadcrumbCat = $localize`Yourself`;
...
 
   constructor(
     private currentContent: CurrentContentService,
+    private currentUserHttpService: CurrentUserHttpService
   ) {
     this.currentContent.current.next(contentInfo({ breadcrumbs: { category: currentUserBreadcrumbCat, path: [], currentPageIdx: -1 } }));
   }
 
+  ngOnInit(): void {
+    this.currentUser$ = this.currentUserHttpService.getProfileInfo();
```
Comment (smadbe) [inline src/app/modules/group/pages/current-user/current-user.component.ts:25]: As I wrote in the spec "No need to fetch the data, it has already been loaded by UserSessionService."

**Review by smadbe — CHANGES_REQUESTED** [review]

File: src/app/shared/http-services/current-user.service.ts (hunk 1 above)
Comment (Iloveall) [reply src/app/shared/http-services/current-user.service.ts]: Fixed

File: src/app/modules/group/pages/current-user/current-user.component.html (hunk 2 above)
Comment (Iloveall) [reply src/app/modules/group/pages/current-user/current-user.component.html]: Fixed

File: src/app/modules/group/pages/current-user/current-user.component.ts (hunk 3 above)
Comment (Iloveall) [reply src/app/modules/group/pages/current-user/current-user.component.ts]: Fixed

File: src/app/modules/group/pages/current-user/current-user.component.html [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <ng-container *ngIf="currentUser$ | async as currentUser">
   <div class="alg-page-header">
-    <div class="alg-page-header-title">{{ currentUser.firstName }} {{ currentUser.lastName }}</div>
+    <div class="alg-page-header-title --large">{{ currentUser.firstName }} {{ currentUser.lastName }}</div>
```
Comment (smadbe) [inline src/app/modules/group/pages/current-user/current-user.component.html:3]: I meant: shouldn't all titles be "large"?

File: src/app/shared/helpers/input-is-not-null-or-undefined.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export function inputIsNotNullOrUndefined<T>(input: null | undefined | T): input is T {
```
Comment (smadbe) [inline src/app/shared/helpers/input-is-not-null-or-undefined.ts:1]: 👍  let's call it `isNotNullNorUndefined` (update file name as well)

**Review by smadbe — CHANGES_REQUESTED** [review]

File: src/app/shared/helpers/input-is-not-null-or-undefined.ts (hunk 5 above)
Comment (Iloveall) [reply src/app/shared/helpers/input-is-not-null-or-undefined.ts]: Fixed

File: src/app/modules/group/pages/current-user/current-user.component.html (hunk 4 above)
Comment (Iloveall) [reply src/app/modules/group/pages/current-user/current-user.component.html]: Fixed

**Review by smadbe — APPROVED** [review]
👍

**Pull request merged**